docs/source/DeveloperGuide/APIs/index.rst

```diff
@@ -0,0 +1,76 @@
+
+    BaseRequestHandler
+    |
+    |\_TornadoService
+    |
+     \_TornadoREST
+
+From the scheme it is intuitively clear that on the basis of BaseRequestHandler it is possible to implement other frameworks.
+
+What is the purpose of this?
+----------------------------
+
+The purpose of this component is to build DIRAC system status access interfaces based on HTTP requests outside the fixed RPC request interface described in TornadoServices.
+
+The main reason for the implementation of this feature is the need to implementation of API for DIRAC Authorization Server based on OAuth2 framework,
+which in turn is an integral part of the implementation of OAuth2 DIRAC authorization.
+So currently, there is only one API implementation for the Framework system, which is :py:class:`~ DIRAC.FrameworkSystem.API.AuthHandler`.
+For the same reason, the ability to authorize with an access token (see https://datatracker.ietf.org/doc/html/rfc6750#section-2) was added to the authorization steps.
+
+How to write APIs
+-----------------
+
+The following is an example of a simple API:
+
+.. literalinclude:: /../../src/DIRAC/Core/Tornado/Server/TornadoREST.py
+    :language: python
+    :start-after: ### Example
+    :end-before: """
+    :caption: Example of system API
+
+.. note:: If you need to implement the interface on a standard 443 https port, you will need to use a balancer, such as nginx
+
+The example described is likely to be sufficient for most writing cases. But here are some additional features, see :py:class:`~ DIRAC.Core.Tornado.Server.private.BaseRequestHandler`:
+
+  - ``USE_AUTHZ_GRANTS`` set the list and order of steps to authorize the request. For example, set ``USE_AUTHZ_GRANTS = ["JWT"]`` to allow access to your endpoint only with a valid access token.
+  - ``AUTH_PROPS`` set the authorization requirements. For example, ``AUTH_PROPS = ['authenticated']`` will allow access only to authenticated users.
+  - in addition to standard S_OK/S_ERROR you can return text, whether the dictionary for example or nothing, the result will be sent with a 200 status.
+  - ``path_<my_method>`` will allow you to consider the path parameters as positional arguments of the target method. For example:
+
+    .. code-block:: python
+
+        # It's allow make request like a GET /user/contacts/Bob
+        path_user = ['[A-z]+', '[A-z]+']
+        def web_user(self, option:str name:str):
+            return Registry.getUserOption(name, option)
+
+  - If your API is complex enough and may include, for example, redirection or additional headers, you can use :py:class:`~ DIRAC.Core.Tornado.Server.private.BaseRequestHandler.TornadoResponse`
+    to add all these necessary things, which is thread-save because TornadoResponse will call your actions outside the thread in which this method is executed:
```

Comment:
```suggestion
    to add all these necessary things, which is thread-safe because TornadoResponse will call your actions outside the thread in which this method is executed:
```